pycbc/distributions/sky_location.py

```diff
@@ -31,5 +34,68 @@ class UniformSky(angular.UniformSolidAngle):
     _default_polar_angle = 'dec'
     _default_azimuthal_angle = 'ra'
 
+class FisherDist():
+    """A distribution that returns a random (ra, dec) angle drawn from the
+    Fisher distribution. Assume that the concentration parameter (kappa)
+    is large so that we can use a Rayleigh distribution about the north
+    pole and rotate it to be centered at the (ra, dec) coordinate mu.
 
-__all__ = ['UniformSky']
+    Assume kappa = 1 / sigma**2
+
+    As in UniformSky, the declination (dec) varies from pi/2 to-pi/2
+    and right ascension (ra) varies from 0 to 2pi. And the angles
+    should be provided in (ra,dec) format in radians (mu_radians=True),
+    rather than factors of pi, or in degrees (mu_radians=False).
+
+    References:
+      * http://en.wikipedia.org/wiki/Von_Mises-Fisher_distribution
+      * http://arxiv.org/pdf/0902.0737v1 (states the Rayleigh limit)
+    """
+    name = 'fisher_dist'
+    _polardistcls = angular.CosAngle
+    _default_polar_angle = 'dec'
+    _default_azimuthal_angle = 'ra'
+
+    def __init__(self, mu_values, kappa, mu_radians=True):
+        self.kappa = kappa
+        if mu_radians:
+            self.mu_values = numpy.array(mu_values[0], mu_values[1])
+        else:
+            self.mu_values = numpy.array(numpy.deg2rad([mu_values[0],
+                                                        mu_values[1]]))
+        self.mu_values = decra2polaz(self.mu_values[1], self.mu_values[0])
+
+    def rvs_polaz(self, size):
+        """
+        Randomly draw multiple samples from the Fisher distribution
+        and returns (polar, azimuthal) angle values.
+        """
+        arr = numpy.array([
+            numpy.random.rayleigh(scale=1./numpy.sqrt(self.kappa),
+                                  size=size),
+            numpy.random.uniform(low=0,
+                                 high=2*numpy.pi,
+                                 size=size)]).reshape((2, size)).T
+        alpha, beta = new_z_to_euler(self.mu_values)
```

Comment:
`alpha` and `beta` are constants fixed at initialization, so you could call`new_z_to_euler()` in `__init__` and store `alpha` and `beta` instead of `mu_values`.